programs/perpetuals/tests/native/tests_suite/lp_token/lp_token_price.rs

```diff
@@ -0,0 +1,230 @@
+        &keypairs[MULTISIG_MEMBER_A],
+        &keypairs[MULTISIG_MEMBER_B],
+        &keypairs[MULTISIG_MEMBER_C],
+    ];
+
+    instructions::test_init(
+        &mut program_test_ctx,
+        upgrade_authority,
+        fixtures::init_params_permissions_full(1),
+        multisig_signers,
+    )
+    .await
+    .unwrap();
+
+    // Initialize and fund associated token accounts
+    {
+        // Alice: mint 100k USDC and 50 ETH
+        {
+            utils::initialize_and_fund_token_account(
+                &mut program_test_ctx,
+                &usdc_mint,
+                &keypairs[USER_ALICE].pubkey(),
+                &keypairs[ROOT_AUTHORITY],
+                utils::scale(100_000, USDC_DECIMALS),
+            )
+            .await;
+
+            utils::initialize_and_fund_token_account(
+                &mut program_test_ctx,
+                &eth_mint,
+                &keypairs[USER_ALICE].pubkey(),
+                &keypairs[ROOT_AUTHORITY],
+                utils::scale(50, ETH_DECIMALS),
+            )
+            .await;
+        }
+    }
+
+    // Set the pool with 50%/50% ETH/USDC liquidity
+    let (pool_pda, _, lp_token_mint_pda, _, custodies_infos) =
+        utils::setup_pool_with_custodies_and_liquidity(
+            &mut program_test_ctx,
+            &keypairs[MULTISIG_MEMBER_A],
+            "FOO",
+            &keypairs[PAYER],
+            multisig_signers,
+            vec![
+                utils::SetupCustodyWithLiquidityParams {
+                    setup_custody_params: utils::SetupCustodyParams {
+                        mint: usdc_mint,
+                        decimals: USDC_DECIMALS,
+                        is_stable: true,
+                        target_ratio: utils::ratio_from_percentage(50.0),
+                        min_ratio: utils::ratio_from_percentage(0.0),
+                        max_ratio: utils::ratio_from_percentage(100.0),
+                        initial_price: utils::scale(1, USDC_DECIMALS),
+                        initial_conf: utils::scale_f64(0.01, USDC_DECIMALS),
+                        pricing_params: None,
+                        permissions: None,
+                        fees: None,
+                        borrow_rate: None,
+                    },
+                    liquidity_amount: utils::scale(15_000, USDC_DECIMALS),
+                    payer: utils::copy_keypair(&keypairs[USER_ALICE]),
+                },
+                utils::SetupCustodyWithLiquidityParams {
+                    setup_custody_params: utils::SetupCustodyParams {
+                        mint: eth_mint,
+                        decimals: ETH_DECIMALS,
+                        is_stable: false,
+                        target_ratio: utils::ratio_from_percentage(50.0),
+                        min_ratio: utils::ratio_from_percentage(0.0),
+                        max_ratio: utils::ratio_from_percentage(100.0),
+                        initial_price: utils::scale(1_500, ETH_DECIMALS),
```

Comment:
unrelated: need to make a PR on our end once I finish staking/multiplier to realign these